tts/aws/src/lib.rs

```diff
@@ -0,0 +1,1153 @@
+};
+use golem_tts::golem::tts::voices::{
+    Guest as VoicesGuest, GuestVoice, GuestVoiceResults, LanguageInfo, Voice, VoiceFilter,
+    VoiceInfo, VoiceResults,
+};
+use log::trace;
+use std::cell::{Cell, RefCell};
+
+mod client;
+mod conversions;
+
+struct PollyVoiceImpl {
+    voice_data: AwsVoice,
+    client: AwsPollyTtsApi,
+}
+
+impl PollyVoiceImpl {
+    fn new(voice_data: AwsVoice, client: AwsPollyTtsApi) -> Self {
+        Self { voice_data, client }
+    }
+}
+
+impl GuestVoice for PollyVoiceImpl {
+    fn get_id(&self) -> String {
+        self.voice_data.id.clone()
+    }
+
+    fn get_name(&self) -> String {
+        self.voice_data.name.clone()
+    }
+
+    fn get_provider_id(&self) -> Option<String> {
+        Some("aws-polly".to_string())
+    }
+
+    fn get_language(&self) -> LanguageCode {
+        self.voice_data.language_code.clone()
+    }
+
+    fn get_additional_languages(&self) -> Vec<LanguageCode> {
+        self.voice_data
+            .additional_language_codes
+            .clone()
+            .unwrap_or_default()
+    }
+
+    fn get_gender(&self) -> VoiceGender {
+        match self.voice_data.gender.to_lowercase().as_str() {
+            "male" => VoiceGender::Male,
+            "female" => VoiceGender::Female,
+            _ => VoiceGender::Neutral,
+        }
+    }
+
+    fn get_quality(&self) -> VoiceQuality {
+        if self
+            .voice_data
+            .supported_engines
+            .contains(&"neural".to_string())
+        {
+            VoiceQuality::Neural
+        } else if self
+            .voice_data
+            .supported_engines
+            .contains(&"generative".to_string())
+        {
+            VoiceQuality::Studio
+        } else {
+            VoiceQuality::Standard
+        }
+    }
+
+    fn get_description(&self) -> Option<String> {
+        Some(format!(
+            "{} voice from AWS Polly",
+            self.voice_data.language_name
+        ))
+    }
+
+    fn supports_ssml(&self) -> bool {
+        true
+    }
+
+    fn get_sample_rates(&self) -> Vec<u32> {
+        vec![8000, 16000, 22050]
+    }
+
+    fn get_supported_formats(&self) -> Vec<AudioFormat> {
+        vec![AudioFormat::Mp3, AudioFormat::Pcm, AudioFormat::OggOpus]
+    }
+
+    fn update_settings(&self, _settings: VoiceSettings) -> Result<(), TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "Voice settings cannot be permanently updated in AWS Polly".to_string(),
+        ))
+    }
+
+    fn delete(&self) -> Result<(), TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "AWS Polly voices cannot be deleted".to_string(),
+        ))
+    }
+
+    fn clone(&self) -> Result<Voice, TtsError> {
+        Err(TtsError::UnsupportedOperation(
+            "Voice cloning not supported for AWS Polly voices".to_string(),
+        ))
+    }
+
+    fn preview(&self, text: String) -> Result<Vec<u8>, TtsError> {
+        let preview_text = if text.len() > 100 {
```

Comment:
Is there a reason for trimming the text here? The `preview` in other providers is not doing it. Should be made consistent (either way)